File: src/Entity/Produit.php
<?php

namespace App\Entity;
use Symfony\Component\Validator\Constraints as Assert;
use App\Repository\ProduitRepository;
use Doctrine\ORM\Mapping as ORM;
use App\Entity\CategorieProduit;
#[ORM\Entity(repositoryClass: ProduitRepository::class)]
class Produit
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column]
    private ?int $id = null;

    #[ORM\Column(length: 255)]
    #[Assert\NotBlank (message:"Please enter product name")]
   
    #[Assert\Length (min:2 , max:30, minMessage:"Your product name must be at least 2 caracteres", maxMessage:"Your product name characters max is 30")]
    private ?string $nom = null;

    #[ORM\Column(length: 255)]
    #[Assert\NotBlank (message:"Please enter product Reference")]
    #[Assert\Length (min:2 , max:30, minMessage:"Your product reference must be at least 2 caracteres", maxMessage:"Your product reference characters max is 30")]
    private ?string $reference = null;

    #[ORM\Column]
    private ?float $prix = null;

    #[ORM\Column(length: 255)]
    #[Assert\NotBlank (message:"Please enter product Color")]
   
    #[Assert\Length (min:2 , max:30, minMessage:"Your product color must be at least 2 caracteres", maxMessage:"Your product color characters max is 30")]
    private ?string $couleur = null;

    #[ORM\Column]
    #[Assert\NotBlank (message:"Please enter product weight")]
    
    private ?float $poids = null;

    #[ORM\Column(length: 255)]
    #[Assert\NotBlank (message:"Please enter product description")]
    private ?string $description = null;

    #[ORM\ManyToOne(inversedBy: 'produits')]
    private ?categorieproduit $id_categorie_produit = null;

    #[ORM\Column(length: 255)]
    private ?string $image = null;

    public function getId(): ?int
    {
        return $this->id;
    }

    public function getNom(): ?string
    {
        return $this->nom;
    }

    public function setNom(string $nom): self
    {
        $this->nom = $nom;

        return $this;
    }

    public function getReference(): ?string
    {
        return $this->reference;
    }

    public function setReference(string $reference): self
    {
        $this->reference = $reference;

        return $this;
    }
    
    public function getPrix(): ?float
    {
        return $this->prix;
    }

    public function setPrix(float $prix): self
    {
        $this->prix = $prix;

        return $this;
    }

    public function getCouleur(): ?string
    {
        return $this->couleur;
    }

    public function setCouleur(string $couleur): self
    {
        $this->couleur = $couleur;

        return $this;
    }

    public function getPoids(): ?float
    {
        return $this->poids;
    }

    public function setPoids(float $poids): self
    {
        $this->poids = $poids;

        return $this;
    }

    public function getDescription(): ?string
    {
        return $this->description;
    }

    public function setDescription(string $description): self
    {
        $this->description = $description;

        return $this;
    }
    
    public function getIdCategorieProduit(): ?categorieproduit
    {
        return $this->id_categorie_produit;
    }

    public function setIdCategorieProduit(?CategorieProduit $id_categorie_produit): self
    {
        $this->id_categorie_produit = $id_categorie_produit;

        return $this;
    }

    public function getImage(): ?string
    {
        return $this->image;
    }

    public function setImage(string $image): self
    {
        $this->image = $image;

        return $this;
    }
}

File: JAVA/src/Entities/produit.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package Entities;

/**
 *
 * @author Admin
 */
public class produit {
   
    private int id;
    private String nom,reference,couleur,description,image;
    private float prix , poids ;
    private categorie_produit categorie;
    private int categorie_id;

    public produit(int id, String nom, String reference, float prix, String couleur, float poids, String description, String image, categorie_produit categorie, int categorie_id) {
        this.id = id;
        this.nom = nom;
        this.reference = reference;
        this.prix = prix;
        this.couleur = couleur;
        this.poids = poids;
        this.description = description;
        this.image = image;
        this.categorie = categorie;
        this.categorie_id = categorie_id;
    }

    public produit(String nom, String reference, float prix, String couleur, float poids, String description, String image, categorie_produit categorie, int categorie_id) {
        this.nom = nom;
        this.reference = reference;
        this.prix = prix;
        this.couleur = couleur;
        this.poids = poids;
        this.description = description;
        this.image = image;
        this.categorie = categorie;
        this.categorie_id = categorie_id;
    }

    public produit(int id, String nom, String reference, float prix, String couleur, float poids, String description, String image, int categorie_id) {
        this.id = id;
        this.nom = nom;
        this.reference = reference;
        this.prix = prix;
        this.couleur = couleur;
        this.poids = poids;
        this.description = description;
        this.image = image;
        this.categorie_id = categorie_id;
    }

    public produit(String nom, String reference, float prix, String couleur, String description, float poids, String image, int categorie_id) {
        this.nom = nom;
        this.reference = reference;
        this.prix = prix;
        this.couleur = couleur;
        this.poids = poids;
        this.description = description;
        this.image = image;
        this.categorie_id = categorie_id;
    }

    public produit(int id, String nom, String reference, float prix, String couleur, float poids, String description, String image) {
        this.id = id;
        this.nom = nom;
        this.reference = reference;
        this.prix = prix;
        this.couleur = couleur;
        this.poids = poids;
        this.description = description;
        this.image = image;
    }
    
   

    public produit() {
    }

    public categorie_produit getCategorie() {
        return categorie;
    }

    public void setCategorie(categorie_produit categorie) {
        this.categorie = categorie;
    }

    public int getCategorie_id() {
        return categorie_id;
    }

    public void setCategorie_id(int categorie_id) {
        this.categorie_id = categorie_id;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getNom() {
        return nom;
    }

    public void setNom(String nom) {
        this.nom = nom;
    }

    public String getReference() {
        return reference;
    }

    public void setReference(String reference) {
        this.reference = reference;
    }

    public float getPrix() {
        return prix;
    }

    public void setPrix(float prix) {
        this.prix = prix;
    }

    public String getCouleur() {
        return couleur;
    }

    public void setCouleur(String couleur) {
        this.couleur = couleur;
    }

    public float getPoids() {
        return poids;
    }

    public void setPoids(float poids) {
        this.poids = poids;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public String getImage() {
        return image;
    }

    public void setImage(String image) {
        this.image = image;
    }

    @Override
    public String toString() {
        return "produit{" + "id=" + id + ", nom=" + nom + ", reference=" + reference + ", prix=" + prix + ", couleur=" + couleur + ", poids=" + poids + ", description=" + description + ", image=" + image + ", categorie=" + categorie + ", categorie_id=" + categorie_id + '}';
    }
}
